common/utils.ts

```diff
@@ -0,0 +1,90 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable no-plusplus */
+import {
+  Card,
+  FlipIndex,
+  TokenColor,
+  TokenBank,
+  tokenColors,
+  cardColors,
+  Player,
+  Deck,
+  Noble,
+  CardBank,
+  CardColor,
+} from './types';
+import { actualCards } from './actualCards';
+import { nobles } from './nobles';
+
+export const shuffle = (array): Array<any> => {
+  for (let i = array.length - 1; i > 0; i--) {
+    const j = Math.floor(Math.random() * (i + 1));
+    [array[i], array[j]] = [array[j], array[i]];
+  }
+  return array;
+};
+
+export const dealLevelDeck = (deck: Card[]): Card[] => {
+  const shuffledDeck = shuffle(deck);
+  shuffledDeck.slice(0, 5).map((elem: Card, i: number) => {
+    elem.flipIndex = i as FlipIndex;
+    return elem;
+  });
+  return shuffledDeck;
+};
+
+export const dealNobles = (playerCount: number): Noble[] => {
+  const shuffledNobles = shuffle(nobles);
+  const nobleCount = playerCount < 4 ? 3 : 4;
+  return shuffledNobles.map((elem: Card, i: number) => {
+    if (i < nobleCount) {
+      elem.flipIndex = i as FlipIndex;
+    } else {
+      elem.flipIndex = undefined;
+    }
+    return elem;
+  });
+};
+
+export const dealGame = (): Deck => {
+  const gameDeck = {};
+  for (let level = 1; level <= 3; level++) {
+    const levelDeck = dealLevelDeck(
+      actualCards.filter((elem) => elem.level === level)
+    );
+    gameDeck[level as keyof Deck] = levelDeck;
+  }
+  return gameDeck as Deck;
+};
+
+export const generateTokenBank = (playerCount: number): TokenBank => {
+  const tokenCount = playerCount < 4 ? 4 : 7;
+  return tokenColors.reduce((acc: TokenBank, elem: TokenColor) => {
+    acc[elem] = elem === 'yellow' ? 7 : tokenCount;
+    return acc;
+  }, {});
+};
+
+export const generateEmptyCardBank = (): CardBank => {
+  return cardColors.reduce((acc: CardBank, elem: CardColor) => {
+    acc[elem] = [];
+    return acc;
+  }, {});
+};
+
+export const generateEmptyTokenBank = (): TokenBank => {
+  return tokenColors.reduce((acc: TokenBank, elem: TokenColor) => {
+    acc[elem] = 0;
+    return acc;
+  }, {});
```

Comment:
Golf. Let's avoid the mutation?

```suggestion
  return tokenColors.reduce((acc: TokenBank, elem: TokenColor) => ({
    [elem]: 0,
    ...acc,
  }), {});
```